### PR TITLE
php: Add autoconf dependency

### PR DESCRIPTION
**Pull request by javian**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
It was discovered that the autoconf dependency was missing as pointed out in #26170 which also made me realise we depend heavily on pecl but its not tested. I'm aware that only the php formula has the test at the moment but I'll add the others once we can confirm that this is the way forward.

### CODE REVIEW

**Review by kabel — APPROVED** [review]
Should probably depend on #26137. The test output also makes me think I should add the channel-update calls to post_install.

File: Formula/php.rb [hunk 1]
```diff
@@ -289,6 +290,8 @@ def plist; <<~EOS
     system "#{sbin}/php-fpm", "-t"
     system "#{bin}/phpdbg", "-V"
     system "#{bin}/php-cgi", "-m"
+    system "#{bin}/pecl", "install", "xdebug"
```
Comment (kabel) [inline Formula/php.rb]: The sandbox isn't going to let you write here.

Comment (javian) [reply Formula/php.rb]: Yeah I noticed that. I thought I could get around it by using `config-set extension_dir` but it didn't seem to work when I tried it. Haven't looked at the pecl code but I'm guess the settings comes from phpize/php-config.

**Comment by javian** [comment]
I removed the test for now since it requires some more work whereas the Autoconf addition is more straightforward.

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
And we're sure `automake` and `libtool` are not needed?

**Comment by ilovezfs** [comment]
(I see libtool is already an indirect dep via unixodbc)

**Comment by fxcoudert** [comment]
@ilovezfs doesn't the new dependency necessitate revision bumps?

**Comment by ilovezfs** [comment]
No, we're doing revision bumps on https://github.com/Homebrew/homebrew-core/pull/26137 so that will take care of it.

**Comment by javian** [comment]
According to the PHP docs `automake` and `libtool` are required dependencies . http://php.net/manual/en/install.unix.php so you are correct. Should I add them as well ?

**Comment by kabel** [comment]
They're only necessary for building from a code repository checkout. Building from a release tarball and running pecl probably shouldn't need them.

**Comment by javian** [comment]
@kabel right, that rings a bell now when we switched back to using php.net releases from the github ones.

**Comment by ilovezfs** [comment]
Thanks again @javian!

**Pull request closed without merging**